src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/handlers/SendEmailsForGoCardlessEventsHandler.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.pay.directdebit.webhook.gocardless.services.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoMandateIdException;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoPaymentIdException;
+import uk.gov.pay.directdebit.events.model.GoCardlessEvent;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.mandate.services.MandateQueryService;
+import uk.gov.pay.directdebit.notifications.services.UserNotificationService;
+import uk.gov.pay.directdebit.payments.exception.PaymentNotFoundException;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentId;
+import uk.gov.pay.directdebit.payments.model.Payment;
+import uk.gov.pay.directdebit.payments.services.PaymentQueryService;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.MANDATES;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYMENTS;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYOUTS;
+
+public class SendEmailsForGoCardlessEventsHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SendEmailsForGoCardlessEventsHandler.class);
+    
+    private static Predicate<GoCardlessEvent> byValidEventsForMandateResourceTypes =
+            event -> MANDATES.equals(event.getResourceType());
+    
+    private static Predicate<GoCardlessEvent> byValidEventsForPaymentsAndPayoutsResourceTypes =
+            event -> Set.of(PAYMENTS, PAYOUTS).contains(event.getResourceType()) &&
+                    Set.of("created", "submitted", "confirmed", "failed", "paid_out", "paid")
```

Comment:
Much like mandates, I think we don’t need to filter by actions here by putting the `if` statement above the payment lookup in the `handlePayment(…)` method.